src/ui/Analytics/Analytics.ts

```diff
@@ -380,6 +380,7 @@ export class Analytics extends Component {
     }
     super.enable();
     this.initializeAnalyticsClient();
+    this.searchInterface.usageAnalytics = this.client;
```

Comment:
Small detail but it would be slightly cleaner to extract this line into a function called `updateSearchInterfaceUAClient()` ,and calling it here and on line 408.

This removes the repetition of implementation, and keeps the calls in the `enable` method at the same layer of abstraction rather than mixing function calls and property assignments.

Same could also be done for lines 406 and 407 grouping them into a function, the idea being that whenever we switch to the NoopClient, we always want to clear pending events. This reduces the chance of a future dev switching to the NoopClient, but forgetting to clear events, and makes the code easier to use.